lib/rules/unit-no-unknown/index.js

```diff
@@ -24,7 +24,12 @@ const ruleMessages = require('../../utils/ruleMessages');
 const validateOptions = require('../../utils/validateOptions');
 const vendor = require('../../utils/vendor');
 const { isRegExp, isString } = require('../../utils/validateTypes');
-const { units } = require('../../reference/units');
+const units = new Set(require('../../reference/units').units); // a copy that is safe to mutate
+
+// `x` as a resolution unit is very often a typo for `px`.
+// By removing it from the set of known units, we can catch those typos.
+// Intentional `x` units are supported by manually checking these in specific functions or properties.
+units.delete('x');
```

Comment:
see : https://github.com/stylelint/stylelint/pull/6906#discussion_r1225933858

Ignoring `x` is very specific to this one rule.
If we want to have a re-usable reference for `resolutionUnits` and if `resolutionUnits` are included in `units`, then we must move the removal of `x` to this rule.

I think this is fine because it makes the exception local to the rule that requires it.

Reply:
see also: https://github.com/stylelint/stylelint/issues/3026#issuecomment-350514289